R/downloadableReactTable.R

```diff
@@ -0,0 +1,166 @@
+    list(
+        shiny::conditionalPanel(
+            # TODO: this condition should be set in the server when
+            # download function is active
+            condition = "output.displayButton",
+            ns        = ns,
+            shiny::span(
+                id    = ns("reactTableButtonDiv"),
+                # TODO: Review this style class when
+                # download function is active to make sure it is consistent with
+                # the table look and feel
+                class = "periscope-downloadable-table-button",
+                style = ifelse(length(downloadtypes) > 0, "", "display:none"),
+                # TODO: Review download button styles and make sure it is
+                # consistent with the new table when download function is active
+                downloadFileButton(ns("reactTableButtonID"),
+                                   downloadtypes,
+                                   hovertext))),
+        reactable::reactableOutput(ns("reactTableOutputID")),
+        # TODO: test this function when table options are passed to
+        # server function
+        shiny::tags$input(
+            id    = ns("reactTableOutputHeight"),
+            type  = "text",
+            class = "shiny-input-container hidden",
+            value = contentHeight),
+        # TODO: test this function when table selection options is active in
+        # server function
+        shiny::tags$input(
+            id    = ns("reactTableSingleSelect"),
+            type  = "text",
+            class = "shiny-input-container hidden",
+            value = singleSelect)
+        )
+}
+
+
+#' downloadableReactTable module server function
+#'
+#' Server-side function for the downloadableReactTableUI.
+#'
+#'
+#' @param id  the ID of the Module's UI element
+#'
+#' @return Rendered react table
+#'
+#' @section Shiny Usage:
+#' This function is not called directly by consumers - it is accessed in
+#' server.R using the same id provided in \code{downloadableReactTableUI}:
+#'
+#' \strong{\code{downloadableReactTable(id)}}
+#'
+#' @seealso \link[periscope2]{downloadableReactTableUI}
+#' @seealso \link[periscope2]{downloadableTableUI}
+#' @seealso \link[periscope2]{downloadableTable}
+#' @seealso \link[periscope2]{downloadFileButton}
+#' @seealso \link[periscope2]{logViewerOutput}
+#' @seealso \link[periscope2]{downloadFile}
+#' @seealso \link[periscope2]{downloadFile_ValidateTypes}
+#' @seealso \link[periscope2]{downloadFile_AvailableTypes}
+#' @seealso \link[periscope2]{downloadablePlot}
+#'
+#' @examples
+#' if (interactive()) {
+#'  library(shiny)
+#'  library(periscope2)
+#'  shinyApp(ui = fluidPage(fluidRow(column(width = 12,
+#'    downloadableReactTableUI("object_id1",
+#'                             downloadtypes = c("csv", "tsv"),
+#'                             hovertext     = "Download the data here!",
+#'                             contentHeight = "300px",
+#'                             singleSelect  = FALSE)))),
```

Comment:
Again, example is using the default, change it to TRUE or remove.

Reply:
done